simplq/src/components/pages/Admin/ShareQueue.jsx

```diff
@@ -26,4 +46,87 @@ const ShareQueue = (props) => {
   );
 };
 
+const ShareQueue = (props) => {
+  const [open, setOpen] = useState(false);
```

Comment:
The component gets re-rendered in the background when we make an API call to the server in the background. (See [here](https://github.com/SimplQ/simplQ-frontend/blob/master/simplq/src/components/pages/Admin/index.jsx#L38))

This causes the popper to close. You can observe this by opening it, and waiting for less than 10s. If it's trivial to get around this, you can try, otherwise leave it as is. This can be handled later more elegantly after https://github.com/SimplQ/simplQ-frontend/pull/507.

Reply:
Ok solved

Reply:
Alright, you mean that it is not helping, right? If not, let's remove the `memo()`

Reply:
I noticed this too, let me see if I can figure out why.

Reply:
It's due to us polling for updates, some logic like this would work, atleast it looks like there are no problems when I tried it.
```suggestion
  let prevData;
  const update = useCallback(() => {
    clearTimeout(timeoutId);
    requestMaker(QueueRequestFactory.get(queueId)).then((data) => {
      if (data && JSON.stringify(prevData) !== JSON.stringify(data)) {
        setTokens(data.tokens);
        ...
```

Basically comparing old data with new one and only changing state if needed. There might be some inbuilt method that react provides for this, not really sure yet.

Reply:
A positive side effect of #507 is that, then one component will fire an action (refresh button, periodic refresh service), and only the component that subscribes to this data gets re-rendered. 

Then we would not have this problem. If we can wait till #507 is done, we can leave it as is and merge this PR. I don't see an urgent need to update main site. Else if we are going to do some fix, let us leave a comment here explaining what and why, so that we remove it post #507 .

Reply:
#507 has its first component ready. To do all the components it will take more time or hands on deck. It should be a priority since it's gonna make a lot of things easier.

Reply:
@mradenovic Can you do one component and let us finalise and merge that. Once that's done, me, @maaverik and others will migrate the remaining components taking your PR as a reference.

I think we can do a lot in the same way. You lead the way and do one sample. Remaining repetitive work, we will take up.

Reply:
Absolutely @daltonfury42  , Anything for me?

Reply:
Yes, can you review #507?